src/common/statistics/StatisticsManager.ts

```diff
@@ -0,0 +1,394 @@
+    const stakeUnlockedEvents =
+      this.allStakeManagerEvents.stakeUnlockedEvents.filter(it => it.returnValues.relayManager === managerAddress)
+    const stakeWithdrawnEvents =
+      this.allStakeManagerEvents.stakeWithdrawnEvents.filter(it => it.returnValues.relayManager === managerAddress)
+    const stakePenalizedEvents =
+      this.allStakeManagerEvents.stakePenalizedEvents.filter(it => it.returnValues.relayManager === managerAddress)
+    const hubAuthorizedEvents =
+      this.allStakeManagerEvents.hubAuthorizedEvents.filter(it => it.returnValues.relayManager === managerAddress)
+    const hubUnauthorizedEvents =
+      this.allStakeManagerEvents.hubUnauthorizedEvents.filter(it => it.returnValues.relayManager === managerAddress)
+    const allEvents =
+      this.allStakeManagerEvents.allEvents.filter(it => it.returnValues.relayManager === managerAddress)
+    return {
+      allEvents,
+      stakeAddedEvents,
+      stakeUnlockedEvents,
+      stakeWithdrawnEvents,
+      stakePenalizedEvents,
+      hubAuthorizedEvents,
+      hubUnauthorizedEvents
+    }
+  }
+
+  async fetchStakeManagerEvents (): Promise<void> {
+    const allEvents = await this.contractInteractor.getPastEventsForStakeManager(allStakeManagerEvents, [], { fromBlock: 1 })
+
+    const stakeAddedEvents = this.extractTransactionInfos<StakeAddedEventInfo>(allEvents, StakeAdded)
+    const stakeUnlockedEvents = this.extractTransactionInfos<StakeUnlockedEventInfo>(allEvents, StakeUnlocked)
+    const stakeWithdrawnEvents = this.extractTransactionInfos<StakeWithdrawnEventInfo>(allEvents, StakeWithdrawn)
+    const stakePenalizedEvents = this.extractTransactionInfos<StakePenalizedEventInfo>(allEvents, StakePenalized)
+    const hubAuthorizedEvents = this.extractTransactionInfos<HubAuthorizedEventInfo>(allEvents, HubAuthorized)
+    const hubUnauthorizedEvents = this.extractTransactionInfos<HubUnauthorizedEventInfo>(allEvents, HubUnauthorized)
+
+    this.allStakeManagerEvents = {
+      allEvents,
+      stakeAddedEvents,
+      stakeUnlockedEvents,
+      stakeWithdrawnEvents,
+      stakePenalizedEvents,
+      hubAuthorizedEvents,
+      hubUnauthorizedEvents
+    }
+  }
+
+  async attemptPing (url: string): Promise<PingResult> {
+    let relayPing: PingResult
+    try {
+      const pingResponse = await this.httpClient.getPingResponse(url)
+      relayPing = { pingResponse }
+    } catch (error) {
+      relayPing = { error }
+    }
+    return relayPing
+  }
+
+  extractTransactionInfos<T extends GNSContractsEvent> (eventsData: EventData[], eventName: EventName): Array<EventTransactionInfo<T>> {
+    return eventsData
+      .filter(eventData => eventData.event === eventName)
+      .map(
+        eventData => {
+          return {
+            eventData,
+            returnValues: eventData.returnValues as T
+          }
+        }
+      )
+  }
+
+  async getAuthorizedHubsAndVersions (stakeManagerEvents: StakeMangerEvents): Promise<ObjectMap<SemVerString>> {
```

Comment:
this is an overkill - and also performance issue: for each entity (relayer), you query all relayhubs it is registered for.
Also, you're stats is for specific hub - so why bother collect into on all hubs
(or vice-versa: if you're collect for all hubs, then start by collecting general hubs info, and then you don't have to query it here)